Source/Charts/Renderers/BarChartRenderer.swift

```diff
@@ -125,9 +125,11 @@ open class BarChartRenderer: BarLineScatterCandleBubbleRenderer
                 if barData.yMin > 0 {
                     heightOffset = CGFloat(offsetAxis.axisMinimum)
                 }
+            }
+            if !offsetAxis._customAxisMax {
```

Comment:
at first I was thinking a fix like this:
```swift
if !offsetAxis._customAxisMin || !offsetAxis._customAxisMax {
                if barData.yMin > 0 {
                    heightOffset = CGFloat(offsetAxis.axisMinimum)
                }
                if barData.yMax < 0 {
                    originYOffset = CGFloat(offsetAxis.axisMaximum)
                }         
            }
```
I think there is a chance when `offsetAxis._customAxisMin` is true, and offsetAxis._customAxisMax is false, you still need to set originYOffset. Like all negative values?

Reply:
After some debugging, it seems that the value of  _customAxisMax/_customAxisMin has nothing to do with calculating offset.